src/Tickets/Commerce/Gateways/Stripe/Settings.php

```diff
@@ -243,80 +243,29 @@ private function get_payment_methods_available() {
 				'currencies' => [ 'GBP' ],
 				'label'      => esc_html__( 'Bacs Direct Debit', 'event-tickets' ),
 			],
-			'bancontact'        => [
-				'currencies' => [ 'EUR' ],
-				'label'      => esc_html__( 'Bancontact', 'event-tickets' ),
-			],
-			'au_becs_debit'     => [
-				'currencies' => [ 'AUD' ],
-				'label'      => esc_html__( 'BECS Direct Debit', 'event-tickets' ),
-			],
-			'boleto'            => [
-				'currencies' => [ 'BRL' ],
-				'label'      => esc_html__( 'Boleto', 'event-tickets' ),
-			],
-			'eps'               => [
-				'currencies' => [ 'EUR' ],
-				'label'      => esc_html__( 'EPS', 'event-tickets' ),
-			],
-			'fpx'               => [
-				'currencies' => [ 'MYR' ],
-				'label'      => esc_html__( 'FPX', 'event-tickets' ),
-			],
 			'giropay'           => [
 				'currencies' => [ 'EUR' ],
 				'label'      => esc_html__( 'Giropay', 'event-tickets' ),
 			],
-			'grabpay'           => [
-				'currencies' => [ 'MYR', 'SGD' ],
-				'label'      => esc_html__( 'GrabPay', 'event-tickets' ),
-			],
-			'ideal'             => [
-				'currencies' => [ 'EUR' ],
-				'label'      => esc_html__( 'iDEAL', 'event-tickets' ),
-			],
 			'klarna'            => [
 				'currencies' => [ 'DKK', 'EUR', 'GBP', 'NOK', 'SEK', 'USD' ],
 				'label'      => esc_html__( 'Klarna', 'event-tickets' ),
 			],
-			'oxxo'              => [
-				'currencies' => [ 'MXN' ],
-				'label'      => esc_html__( 'OXXO', 'event-tickets' ),
-			],
-			'p24'               => [
-				'currencies' => [ 'EUR', 'PLN' ],
-				'label'      => esc_html__( 'P24', 'event-tickets' ),
-			],
 			'acss_debit'        => [
 				'currencies' => [ 'CAD', 'USD' ],
 				'label'      => esc_html__( 'Pre-authorized debit in Canada', 'event-tickets' ),
 			],
-			'sepa_debit'        => [
-				'currencies' => [ 'EUR' ],
-				'label'      => esc_html__( 'SEPA debit', 'event-tickets' ),
-			],
-			'sofort'            => [
-				'currencies' => [ 'EUR' ],
-				'label'      => esc_html__( 'Sofort', 'event-tickets' ),
-			],
-			'wechat_pay'        => [
-				'currencies' => [
-					'AUD',
-					'CAD',
-					'CHF',
-					'CNY',
-					'DKK',
-					'GBP',
-					'HKD',
-					'JPY',
-					'NOK',
-					'SEK',
-					'SGD',
-					'USD',
-				],
-				'label'      => esc_html__( 'WeChat Pay', 'event-tickets' ),
-			],
 		];
+
+		$tickets_plus_methods = [];
+
+		if ( $this->is_licensed_plugin() ) {
+			if ( class_exists( 'TEC\Tickets_Plus\Commerce\Gateways\Stripe\Settings' ) ) {
+				$tickets_plus_methods = tribe( \TEC\Tickets_Plus\Commerce\Gateways\Stripe\Settings::class )->get_payment_methods_available();
```

Comment:
I feel this should be coming in a filter. We're moving out of the dynamic of how we extend things from the premium plugin on top of the base plugin.

I understand the point that they can add their own function with the filter. But as I mentioned before, it's open source software and they could always modify things to add the payment methods. Or even modify `is_current_license_valid ` to return always true.